### DISCUSSION TITLE
Trouble finding necessary libraries

### DISCUSSION THREAD

**Discussion by Drayux** [stepmania/stepmania#2245] [Q&A] [answered]
Hey! I'm in the process of trying to get Stepmania to run on my Arch Linux install. The current AUR package seems to be having troubles compiling, so I figured it a good opportunity to manually install.

I've downloaded the 5.0.12 release from the releases page on this repo and extracted the archive. I was running into issues where the executable cannot find certain libraries. Many of which I was able to find a package that supplied them, however `libva.so.1` and `libpcre.so.3` are still unaccounted for. I cannot find a package that provides these libraries. (Determined by running `ldd` on the executable.)

Any help as to how to get my Stepmania up and running is appreciated!

**Comment by takoWorkshop** [comment, marked as answer]
Hello, very late response I know, but I've managed to solve the problem, so I figured I should leave it here.

In my case, the libraries were included with my operating system (Garuda Spizaetus), but the links to them that Stepmania was looking for did not exist. You can search for the libraries in question with 

`ls /usr/lib | grep libva` and `ls /usr/lib | grep libpcre` in the terminal.

Once you've verified that the libraries are present, you can create the proper symbolic links to them with 

`sudo ln -s /usr/lib/libva.so /usr/lib/libva.so.1`, and
`sudo ln -s /usr/lib/libpcre.so /usr/lib/pcre.so.3`.

Of course on different distributions the libraries may be in slightly different locations, so a bit of digging around may be required. After creating the links though, Stepmania launches with no problems.

**Reply by Drayux** [reply]
Hey I really appreciate this response! Truthfully I've since completely forgotten that I'd made this post to begin with! As a corollary, I've also since forgotten the details of my trouble and what I did to fix them. (Though admittedly I think it was just to build ITGMania instead.)

That said, I've definitely since then learned much more about the Linux operating system, and I fully agree with your solution. I'd bet money that I was missing those symlinks back then. Absent the extra detail I'd once had, I'd say that it's safe to call this a solution! :)
